tests/client/test_client_peer.py

```diff
@@ -52,7 +50,6 @@ async def test_init_client_peer(
 async def test_init_client_peer_with_alias(
     peer_listener: PeerListener,
     peer_manager: PeerManager,
-    test_endpoint: list[Client],
```

Comment:
Maybe we can keep and replace `request_handler_protocol = protocol_factory()` with a protocol that does raw read/writes but that might be a bit complex to write a whole new one